scripts/generate-swagger-ui.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+
+set -e
+
+[[ -z "$SWAGGER_UI_VERSION" ]] && echo "missing \$SWAGGER_UI_VERSION" && exit 1
+
+SWAGGER_UI_GIT="https://github.com/swagger-api/swagger-ui.git"
+CACHE_DIR="./.cache/swagger-ui/$SWAGGER_UI_VERSION"
+GEN_DIR="./third_party/OpenAPI"
+
+escape_str() {
+  echo "$1" | sed -e 's/[]\/$*.^[]/\\&/g'
+}
+
+# do caching if there's no cache yet
+if [[ ! -d "$CACHE_DIR" ]]; then
+  mkdir -p "$CACHE_DIR"
+  tmp="$(mktemp -d)"
+  git clone --depth 1 --branch "$SWAGGER_UI_VERSION" "$SWAGGER_UI_GIT" "$tmp"
+  cp -r "$tmp/dist/" "$CACHE_DIR"
+  cp -r "$tmp/LICENSE" "$CACHE_DIR"
+  rm -rf "$tmp"
+fi
+
+# populate swagger.json
+tmp="    urls: ["
+for i in $(find "$GEN_DIR" -name "*.swagger.json"); do
+  escaped_gen_dir="$(escape_str "$GEN_DIR/")"
+  path="$(echo $i | sed -e "s/$escaped_gen_dir//g")"
+  tmp="$tmp{\"url\":\"$path\",\"name\":\"$path\"},"
+done
+# delete last characters from $tmp
+tmp=$(echo "$tmp" | sed 's/.$//')
+tmp="$tmp],"
+
+# recreate swagger-ui, delete all except swagger.json
+find "$GEN_DIR" -type f -not -name "*.swagger.json" -delete
+mkdir -p "$GEN_DIR"
+cp -r "$CACHE_DIR/" "$GEN_DIR"
+
+# replace the default URL
+line="$(cat "$GEN_DIR/swagger-initializer.js" | grep -n "url" | cut -f1 -d:)"
+escaped_tmp="$(escape_str "$tmp")"
+sed -i '' -e "$line s/^.*$/$escaped_tmp/" "$GEN_DIR/swagger-initializer.js"
```

Comment:
This `sed` line doesn't work on my linux distro (GNU sed 4.8). I'm guessing you tested this on a mac? According to https://stackoverflow.com/a/4247319 we should be able to use 
```suggestion
sed -i'' -e "$line s/^.*$/$escaped_tmp/" "$GEN_DIR/swagger-initializer.js"
```
To support both macos 10.9+ and GNU.

Reply:
you're right, I'm on mac, and your suggestion don't work on me, lemme try to do some switching here

Reply:
my bad, that one's work fine, but it creates another file on my side `$GEN_DIR/swagger-initializer.js-e`